sdk/examples/wallet/spammer.rs

```diff
@@ -0,0 +1,198 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! In this example we will spam transactions from multiple threads simultaneously to our own address.
+//!
+//! `cargo run --example threads --release`
+
+use std::{env::var, time::Duration};
+
+use iota_sdk::{
+    client::{
+        constants::SHIMMER_COIN_TYPE,
+        request_funds_from_faucet,
+        secret::{mnemonic::MnemonicSecretManager, SecretManager},
+    },
+    wallet::{account::types::AccountAddress, Account, ClientOptions, Result, SendAmountParams, Wallet},
+};
+use tokio::{task::JoinSet, time::sleep};
+
+// The account alias used in this example.
+const ACCOUNT_ALIAS: &str = "spammer";
+// The number of spamming rounds.
+const NUM_ROUNDS: usize = 1000;
+// The amount to send in each transaction.
+const SEND_AMOUNT: u64 = 1_000_000;
+// The number of concurrent transactions.
+const NUM_CONCURRENT_TXS: usize = 32;
+
+#[tokio::main(flavor = "multi_thread")]
+async fn main() -> Result<()> {
+    // This example uses secrets in environment variables for simplicity which should not be done in production.
+    dotenvy::dotenv().ok();
+
+    println!(
+        "{} cores detected. Running spammer with {NUM_CONCURRENT_TXS} concurrent transactions...",
+        num_cpus::get(),
+    );
+
+    // Restore wallet from a mnemonic phrase.
+    let client_options = ClientOptions::new().with_node(&std::env::var("NODE_URL").unwrap())?;
+    let secret_manager =
+        MnemonicSecretManager::try_from_mnemonic(&std::env::var("NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1").unwrap())?;
+    let wallet = Wallet::builder()
+        .with_secret_manager(SecretManager::Mnemonic(secret_manager))
+        .with_storage_path(&var("WALLET_DB_PATH").unwrap())
+        .with_client_options(client_options)
+        .with_coin_type(SHIMMER_COIN_TYPE)
+        .finish()
+        .await?;
+
+    // Ensure there's some base coin balance
+    let account = get_or_create_account(&wallet, ACCOUNT_ALIAS).await?;
+    let addresses = generate_max_addresses(&account, NUM_CONCURRENT_TXS).await?;
```

Comment:
No need to generate multiple addresses, this will only make syncing slower

Reply:
Changed some time a ago already ... please resolve :trollface: